Source/engine/random.hpp

```diff
@@ -136,6 +136,83 @@ class DiabloGenerator {
 	}
 };
 
+/** Adapted from https://prng.di.unimi.it/xoshiro128plusplus.c */
+class xoshiro128plusplus {
+private:
+	static uint32_t rotl(const uint32_t x, int k)
+	{
+		return (x << k) | (x >> (32 - k));
+	}
+
+	uint32_t s[4];
+
+public:
+	void seed(uint64_t value)
+	{
+		uint64_t rand1 = next_split(value);
+		uint64_t rand2 = next_split(rand1);
+		s[0] = static_cast<uint32_t>(rand1);
+		s[1] = static_cast<uint32_t>(rand2);
+		s[2] = static_cast<uint32_t>(rand1 >> 32);
+		s[3] = static_cast<uint32_t>(rand2 >> 32);
+	}
+
+	void seed(uint32_t value)
+	{
+		s[0] = next_split(value);
+		s[1] = next_split(s[0]);
+		s[2] = next_split(s[1]);
+		s[3] = next_split(s[2]);
```

Comment:
If you define a splitmix32 class with a generate() function along the lines of a SeedSequence you can implement this function as
```suggestion
		SplitMix32 seedSequence{ value }; // example SplitMix32 implementation below

		seedSequence.generate(s, s + 4);
```

```cpp
// Written in 2016 by Kaito Udagawa
// Released under CC0 <http://creativecommons.org/publicdomain/zero/1.0/>
// Ported to C++ from https://github.com/umireon/my-random-stuff/blob/e7b17f992955f4dbb02d4016682113b48b2f6ec1/xorshift/splitmix32.c
class SplitMix32 {
	uint32_t state;

public:
	SplitMix32(uint32_t state) : state(state) {}

	// MurmurHash3 fmix32 with adding GOLDEN_GAMMA [1]
	// [1]: Guy L. Steele, Jr., Doug Lea, and Christine H. Flood. 2014. Fast splittable pseudorandom number generators. In Proceedings of the 2014 ACM International Conference on Object Oriented Programming Systems Languages & Applications (OOPSLA '14). ACM, New York, NY, USA, 453-472.
	uint32_t next()	{
		uint32_t z = (state += 0x9e3779b9);
		z = (z ^ (z >> 16)) * 0x85ebca6b;
		z = (z ^ (z >> 13)) * 0xc2b2ae35;
		return z ^ (z >> 16);
	}

	void generate(uint32_t* begin, const uint32_t* end) {
		while (begin != end) {
			*begin = next();
			++begin;
		}
	}
};
```